### PR TITLE
Basic viewer implementation

### PR DESCRIPTION
**Pull request by kylelwm**
# Changelog
#### WorldScope can now view streams
- ##### Steps to view stream
  - ###### Navigate into Main Activity
  - ###### Tap on the demo stream
  - ###### Wait for stream to load
  - ###### Enjoy watching the stream!
#### Details of changes
- ##### Added assets for Play icon and Record icon
- ##### Updated the icons for the button in MainActivity and RecordActivity to a video camera icon and record icon respectively
- ##### Imported Exoplayer into WorldScope Android project via Gradle
- ##### Changed RTMP link in RecordActivity to the updated one
- ##### Changed tabs header in MainActivity to the same one as the mock up
- ##### Preliminary work for viewing a stream using ExoPlayer (Important!)
  - ###### LitmusPlayer.java - WorldScope's instance of a DASH Stream Player made possible by the ExoPlayer
  - ###### DashRendererBuilder.java - Contains the stream building logic, converts a mpd resource into a video and audio track renderer playable by LitmusPlayer
  - ###### ViewActivity.java - Contains basic viewing logic. Instantiates LitmusPlayer and passes the mpd link and the surface from the UI into it for playback later
  - ###### activity_view.xml - Contains the surface for playback and a button for pausing and playing

### CODE REVIEW

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/MainActivity.java [hunk 1]
```diff
@@ -128,11 +128,11 @@ public int getCount() {
...
+                    return "Popular";
                 case 1:
-                    return "SECTION 2";
+                    return "Latest";
                 case 2:
-                    return "SECTION 3";
+                    return "Friends";
             }
             return null;
         }
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/MainActivity.java:138]: "Friends" referring to the people a user subscribed to? We don't have 'Friends' in our project glossary.

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/LitmusPlayer.java [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    public void readyToPushSurface(int indicator){
+
+        if(indicator == SURFACE_READY) {
+            surfaceLock = false;
+        }
+        else if(indicator == RENDERER_READY) {
+            rendererLock = false;
+        }
+
+        Log.i(TAG, surfaceLock + " " + rendererLock);
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/LitmusPlayer.java:116]:
``` java
if (a) {
  ....
} else if (b) {
  ...
}
```

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/LitmusPlayer.java [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+        // Complete preparation
+        this.videoRenderer = renderers[TYPE_VIDEO];
+        this.audioRenderer = (MediaCodecAudioTrackRenderer) renderers[TYPE_AUDIO];
+        this.codecCounters = videoRenderer instanceof MediaCodecTrackRenderer
+                ? ((MediaCodecTrackRenderer) videoRenderer).codecCounters
+                : renderers[TYPE_AUDIO] instanceof MediaCodecTrackRenderer
+                ? ((MediaCodecTrackRenderer) renderers[TYPE_AUDIO]).codecCounters : null;
+        Log.i(TAG, "Preparing player");
+        this.player.prepare(this.videoRenderer, this.audioRenderer);
+    }
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/LitmusPlayer.java:229]: Using If-else statements would improve visibility. (If this section was not auto-gen)

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/LitmusPlayer.java [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    public static final int TRACK_DISABLED = ExoPlayer.TRACK_DISABLED;
+    public static final int TRACK_DEFAULT = ExoPlayer.TRACK_DEFAULT;
+
+    // Locks and conditions for push function to execute
+    final Lock pushLock = new ReentrantLock();
+    final Condition noSurface = pushLock.newCondition();
+    final Condition noRenderers = pushLock.newCondition();
+
+    // Handler given to DashRendererBuilder
+    private Handler mainHandler;
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/LitmusPlayer.java:84]: You used surfaceLock and rendererLock instead  of Conditions and Locks used to solve the async issue?

**Comment by kylelwm** [comment]
Thanks for reviewing! =)

**Pull request merged**